### PR TITLE
Refactoring/refactoring mvp

### PR DESCRIPTION
**Pull request by alexbuchkin**
DO NOT MERGE THIS PULL REQUEST!
tried to do some refactoring, need to continue

### CODE REVIEW

File: bot/bot.py [hunk 1]
```diff
@@ -462,50 +459,48 @@ def join_event_handler(
...
-                                               '456.23(дробная число, целая часть от дробной отделяется точкой. '
+            update.effective_chat.send_message('Потраченная сумма вводится в формате: 123(целое число). '
                                                'Попробуй еще раз')
             return
 
-        self.bot_.expenses[user_id]["sum"] = expense_sum
-        expense = self.bot_.expenses[user_id]
-        expense_id = self.bot_.splitwise.add_expense(expense['name'], user_id, expense['token'], expense['sum'])
-        self.bot_.expenses[user_id]["id"] = expense_id
+        expense_id = self.bot_.splitwise.add_expense(self.bot_.expenses[user_id])
```
Comment (karabutov) [inline bot/bot.py]: Hello, Alex! How are you doing? Hope you don't mind specifying expense_sum before persisting it to the db

File: bot/buttons.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+
 
-def get_user_buttons(users: list):
-    keyboard = list()
-    for user in users:
-        line = [InlineKeyboardButton(user['name'], callback_data=user['id'])]
-        keyboard.append(line)
+def get_user_buttons(users: List[User]) -> InlineKeyboardMarkup:
+    keyboard = [
+        [InlineKeyboardButton(user.name, callback_data=user.id)]
```
Comment (karabutov) [inline bot/buttons.py:11]: I think we should use telegram username(update.effective_user.username) instead of just name

File: bot/bot.py [hunk 3]
```diff
@@ -110,15 +114,12 @@ def start_handler(
         name = update.effective_user.full_name
```
Comment (karabutov) [inline bot/bot.py]: Let's use update.effective_user.username instead update.effective_user.full_name

File: bot/bot.py (hunk 1 above)
Comment (alexbuchkin) [reply bot/bot.py]: Hello, Dmitry! Thank you very much for this issue! I will fix it as soon as possible.

**Comment by rusgeorge** [comment]
incredible work
unbelievable solutions

**Pull request closed without merging**